### PR TITLE
feat(frontend): add confirmation before update docs published status

### LINKED ISSUE #307: Administration - Ajouter un confirmation dans le processus de publication / dépublication
On souhaite pouvoir introduire une confirmation de l'utilisateur lors du changement de status d'un document.
Il peut aussi être utile de prévoir un toast avec la possibilité d'annuler l'opération (un peu comme dans gmail, pour les message dans la corbeille)

### PR DESCRIPTION
**Pull request by lionelB**
- add confirmation before changing document published status
- allow select documents on multiple page
 
closes #307

### CODE REVIEW

**Review by m-maillot — COMMENTED** [review]
I'm not comfortable with this implementation 😄 

Why did you not implement the updatePublication at the root? I think the `List` can be a simple component that shows a list. I don't understand why `SearchFilter` is not only in the root component. Can we take a moment to explain me your implementation? (Perhaps, I miss something)

File: targets/frontend/src/components/documents/Container.js [hunk 1]
```diff
@@ -0,0 +1,164 @@
...
+      setSelectedItems({});
+      const query = { ...filterValues, page: 0 };
+      router.push({ pathname: router.route, query }, undefined, {
+        shallow: true,
+      });
+    },
+    [router, setSelectedItems]
+  );
+
+  const [result] = useQuery({
```
Comment (m-maillot) [inline targets/frontend/src/components/documents/Container.js:44]: I don't understand how the list is refreshed when we update the publication status of documents. For the search, we push a new path in the router, the component is rerender and useQuery is performed. But when we change the publication status, we call the `updatePublicationStatus` which doesn't trigger a new call to fetch all documents 🤔

**Review by m-maillot — COMMENTED** [review]
I don't understand how we refresh the list when we change the publication's status. otherwise, it's looking good to me :+1:

File: targets/frontend/src/components/documents/Container.js (hunk 1 above)
Comment (lionelB) [reply targets/frontend/src/components/documents/Container.js:44]: urql update it's cache since there is a mutation on documents data

Comment (m-maillot) [reply targets/frontend/src/components/documents/Container.js:44]: Ok. (I didn't see that on their documentation, probably miss it 🤷 😕)

**Pull request merged**